modes/taxi.md

```diff
@@ -0,0 +1,82 @@
+# Taxi Mode
+
+## Taxi Vehicle States
+
+| `state`           | In PROW? | Description                                                                                                                                                                    |
+|-------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `removed`         | no       | Vehicle is explicitly removed from the PROW. Examples include: at the shop receiving maintenance, vehicle moved to another fleet                                               |
+| `available`       | yes      | Available for-hire by a passenger.                                                                                                                                             |
+| `non_operational` | yes      | Not available for-hire.  Example: driver's shift ends and vehicle goes out of service                                                                                          |
+| `reserved`        | yes      | Reserved via Provider's app, waiting to be picked up by a rider.                                                                                                               |
```

Comment:
How can we account for instances where a rider has requested a vehicle, but that request was never served?  The state machine is vehicle-centric, but in this case a request would never be associated with a vehicle.

Reply:
Interesting -- so you're thinking of a case where there is a request for a reservation, but no `reserve` event ever (dispatch never allocated a vehicle for the rider at all)? If that's what you're thinking, I bet we could just have trip metadata without any corresponding events!

Reply:
Yes exactly - having it in trip metadata makes sense!